exporter.php

```diff
@@ -1,257 +1,51 @@
+ * Campaign Exporter Functionality
  *
  * @package P4MT
  */
 
 global $wpdb, $post;
-$defaults = [
-	'content'    => 'all',
-	'author'     => false,
-	'category'   => false,
-	'start_date' => false,
-	'end_date'   => false,
-	'status'     => false,
-];
-$args     = wp_parse_args( $post, $defaults );
+
+require_once 'exporter-helper.php';
+
 define( 'CMX_VERSION', '1.0' );
 $sitename = sanitize_key( get_bloginfo( 'name' ) );
 if ( ! empty( $sitename ) ) {
 	$sitename .= '.';
 }
-$filename = $sitename . gmdate( 'Ymdhis' ) . '.xml';
 
-header( 'Content-Description: File Transfer' );
-header( 'Content-Disposition: attachment; filename=' . $filename );
-header( 'Content-Type: text/xml; charset=' . get_option( 'blog_charset' ), true );
+// Sanitize input.
+$post_ids = filter_input( INPUT_GET, 'post', FILTER_SANITIZE_STRING );
 
-if ( 'all' !== $args['content'] && post_type_exists( $args['content'] ) ) {
-	$ptype = get_post_type_object( $args['content'] );
+$post_ids = explode( ',', $post_ids );
 
-	if ( ! $ptype->can_export ) {
-		$args['content'] = 'post';
+if ( 1 === count( $post_ids ) ) {
+	$slug = get_post_field( 'post_name', $post_ids[0] );
+	if ( ! $slug ) {
+		$slug = 'P4-Campaign';
 	}
-
-	$where = $wpdb->prepare( "{$wpdb->posts}.post_type = %s", $args['content'] );
+	// Add campaign slug in XML filename.
+	$filename = $sitename . $slug . '.' . gmdate( 'Y-m-d' ) . '.xml';
 } else {
-	$post_types = get_post_types( [ 'can_export' => true ] );
-	$esses      = implode( ',', array_fill( 0, count( $post_types ), '%s' ) );
-	$where      = $wpdb->prepare( "{$wpdb->posts}.post_type IN ( %s )", $esses );
-}
-
-if ( $args['status'] && ( 'post' === $args['content'] || 'page' === $args['content'] ) ) {
-	$where .= $wpdb->prepare( " AND {$wpdb->posts}.post_status = %s", $args['status'] );
-} else {
-	$where .= " AND {$wpdb->posts}.post_status != 'auto-draft'";
-}
-
-$join = '';
-if ( $args['category'] && 'post' === $args['content'] ) {
-	$term = term_exists( $args['category'], 'category' );
-	if ( $term ) {
-		$join   = "INNER JOIN {$wpdb->term_relationships} ON ({$wpdb->posts}.ID = {$wpdb->term_relationships}.object_id)";
-		$where .= $wpdb->prepare( " AND {$wpdb->term_relationships}.term_taxonomy_id = %d", $term['term_taxonomy_id'] );
-	}
-}
-
-if ( 'post' === $args['content'] || 'page' === $args['content'] ) {
-	if ( $args['author'] ) {
-		$where .= $wpdb->prepare( " AND {$wpdb->posts}.post_author = %d", $args['author'] );
-	}
-
-	if ( $args['start_date'] ) {
-		$where .= $wpdb->prepare( " AND {$wpdb->posts}.post_date >= %s", date( 'Y-m-d', strtotime( $args['start_date'] ) ) );
-	}
-
-	if ( $args['end_date'] ) {
-		$where .= $wpdb->prepare( " AND {$wpdb->posts}.post_date < %s", date( 'Y-m-d', strtotime( '+1 month', strtotime( $args['end_date'] ) ) ) );
-	}
+	// Add "P4-Campaign" word in XML filename to distinguish between normal WP export and P4 campaign export.
+	$filename = $sitename . 'P4-Campaign.' . gmdate( 'Y-m-d' ) . '.xml';
```

Comment:
Maybe now we have the name for exports of single campaigns, we can make this plural?